tests/test_gradient.py

```diff
@@ -0,0 +1,295 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+from smrf.data import loadTopo
+from smrf.utils import gradient
+
+from tests.test_configurations import SMRFTestCase
+
+
+class TestGradient(SMRFTestCase):
+
+    def gen_dem_nw(self, dem_size=10):
+        dem = np.tile(range(10), (10, 1)).transpose()
+
+        for i in range(dem_size):
+            dem[i, :] = np.arange(i, i+dem_size)
+        return dem
+
+    def gen_dem_sw(self, dem_size=10):
+        dem = np.tile(range(10), (10, 1)).transpose()
+
+        for i in range(1, 1+dem_size):
+            dem[-i, :] = np.arange(i, i+dem_size)
+        return dem
+
+    def gen_dem_se(self, dem_size=10):
+        dem = np.tile(range(10), (10, 1)).transpose()
+
+        for i in range(1, 1+dem_size):
+            dem[-i, :] = np.arange(i+dem_size, i, -1)
+        return dem
+
+    def gen_dem_ne(self, dem_size=10):
+        dem = np.tile(range(10), (10, 1)).transpose()
+
+        for i in range(dem_size):
+            dem[i, :] = np.arange(i+dem_size, i, -1)
+        return dem
+
+    def test_gradient_d4_funcs(self):
```

Comment:
Minor feedback that I have and been taught about unit testing is to break up your assertion into multiple cases, once the method gets too long. 
Has some advantages, when things start to fail. [This stack exchange](https://softwareengineering.stackexchange.com/questions/260873/how-much-should-i-break-up-my-unit-tests) names a few and shows some ideas on how to break It up.

For this instance, I would propose to have cases for each direction (north, south, ...).

Reply:
Makes sense, when we get to reviewing this PR I'll make those changes. Not sure why I did it this way!

Reply:
Broke these out into separate tests

Reply:
AWSM ... ehh .. SMRF .. you know what I mean :-)